tests/integration/study_data_blueprint/test_link.py

```diff
@@ -19,6 +20,117 @@
+        if study_type == "variant":
+            study_id = preparer.create_variant(study_id, name="Variant 1")
+
+        area1_id = preparer.create_area(study_id, name="Area 1")["id"]
+        area2_id = preparer.create_area(study_id, name="Area 2")["id"]
+        client.post(f"/v1/studies/{study_id}/links", json={"area1": area1_id, "area2": area2_id, "hurdlesCost": True})
+        res = client.put(
+            f"/v1/studies/{study_id}/links",
+            json={"area1": area1_id, "area2": area2_id, "colorr": 150},
+        )
+
+        assert res.status_code == 200
+        expected = {
+            "area1": "area 1",
+            "area2": "area 2",
+            "assetType": "ac",
+            "colorb": 112,
+            "colorg": 112,
+            "colorr": 150,
+            "displayComments": True,
+            "filterSynthesis": "hourly, daily, weekly, monthly, annual",
+            "filterYearByYear": "hourly, daily, weekly, monthly, annual",
+            "hurdlesCost": True,
+            "linkStyle": "plain",
+            "linkWidth": 1.0,
+            "loopFlow": False,
+            "transmissionCapacities": "enabled",
+            "usePhaseShifter": False,
+        }
+        assert expected == res.json()
+
+        # Test update link same area
+
+        res = client.put(
+            f"/v1/studies/{study_id}/links",
+            json={
+                "area1": area1_id,
+                "area2": area1_id,
+                "hurdlesCost": False,
+            },
+        )
+        assert res.status_code == 422
+        expected = {
+            "description": "Cannot create a link that goes from and to the same single area: area 1",
+            "exception": "LinkValidationError",
+        }
+        assert expected == res.json()
+
+        # Test update link with non existing area
+
+        res = client.put(
+            f"/v1/studies/{study_id}/links",
+            json={
+                "area1": area1_id,
+                "area2": "id_do_not_exist",
+                "hurdlesCost": False,
+            },
+        )
+        assert res.status_code == 422
+        expected = {
+            "description": "The link area 1 -> id_do_not_exist is not present in the study",
+            "exception": "LinkValidationError",
+        }
+        assert expected == res.json()
+
+        # Test update link fails when given wrong parameters
+        if study_type == "raw":
+            res = client.post(
+                f"/v1/studies/{study_id}/commands",
+                json=[
+                    {
+                        "action": "update_link",
+                        "args": {
+                            "area1": area1_id,
+                            "area2": area2_id,
+                            "parameters": {"hurdles-cost": False, "wrong": "parameter"},
+                        },
+                    }
+                ],
+            )
+            assert res.status_code == 500
+            expected = {
```

Comment:
You shouldn't check the entire content as if pydantic changes its error msg we'll have to change this. You could just check our msg: "Unexpected exception occurred when trying to apply command CommandName.UPDATE_LINK: 1 validation error for LinkInternal"